src/kerchunk/gran_to_cat.py

```diff
@@ -0,0 +1,197 @@
+        warnings.simplefilter("ignore")
+        refs = SingleHdf5ToZarr(BytesIO(data), s3_url).translate()
+    nc = Dataset("t", memory=data)
+    attrs = ujson.loads(refs["refs"][".zattrs"])
+    # Inline coordinate arrays directly using delta encoding, this way
+    # we don"t need to read them directly from netCDF in the future
+    for c in ["x", "y"]:
+        val = nc[c][:]
+        z = zarr.array(val, compressor=zarr.Blosc(cname="zstd", clevel=5),
+                       filters=[zarr.Delta(val.dtype)])
+        refs["refs"][f"{c}/.zarray"] = z.store[".zarray"].decode()
+        refs["refs"][f"{c}/0"] = "base64:" + base64.b64encode(z.store["0"]).decode()
+    # Kerchunk can"t handle these structs properly so these are converted to
+    # global attributes
+    for variable in ["mapping", "img_pair_info"]:
+        obj = nc[variable]
+        for attr in obj.ncattrs():
+            val = getattr(obj, attr)
+            if hasattr(val, "item"):
+                val = val.item()
+            attrs[attr] = val
+        del refs["refs"][variable+"/.zarray"]
+    refs["refs"][".zattrs"] = ujson.dumps(attrs)
+    return refs
+
+
+class DuckVariable:
+    """Simple mock xarray variable"""
+    def __init__(self, name, attrs, chunks, dims, shape):
+        self.name = name
+        self.attrs = attrs
+        self.chunks = chunks
+        self.dims = dims
+        self.shape = shape
+        
+    @property
+    def chunksize(self):
+        return self.chunks
+        
+    @property
+    def data(self):
+        return self
+
+class DuckDataset:
+    """Mock lightweight dataset object from kerchunk"""
+    def __init__(self, refs: Dict[str, Any]):
+        r = refs["refs"]
+        vnames = [k.split("/")[0] for k in r.keys() if "/" in k]
+        self.variables = {}
+        self.dims = ["y", "x"]
+        self.coords = self.dims
+        for v in vnames:
+            attrs = ujson.loads(r.get(f"{v}/.zattrs", "{}"))
+            zarray = ujson.loads(r[f"{v}/.zarray"])
+            shape = zarray["shape"]
+            chunks = zarray["chunks"]
+            dims = attrs.get("_ARRAY_DIMENSIONS")
+            if dims is None:
+              continue
+            dims = dict(zip(dims, zarray["shape"]))
+            self.variables[v] = DuckVariable(v, attrs, chunks, dims, shape)
+            
+    def __getitem__(self, attr: str):
+        return self.variables[attr]
+        
+    def __getattr__(self, attr: str):
+        if attr in self.variables:
+            return self.variables[attr]
+        return super().__getattr__(attr)
+
+
+def coord_from_ref(ref: Dict[str, Any], dim: str):
+    data = ref["refs"][f"{dim}/0"][6:]
+    blosc = zarr.Blosc("zstd")
+    delta = zarr.Delta(dtype=np.float64)
+    decoders = [base64.b64decode, blosc.decode, delta.decode]
+    for decode in decoders:
+        data = decode(data)
+    return data
+    
+        
+def geom_from_refs(refs: Dict[str, Any]):
+    def maybe_scalar(g):
```

Comment:
I think this would be better called `force_scaler` or `ensure_scalar` as you're extracting a scalar out of a tuple here.